Failed to build (Release) the latest 'main' branch with MGHIO enabled

```
[ 96%] Building CXX object Modules/Remote/MGHIO/src/CMakeFiles/MGHIO.dir/itkMGHImageIOFactory.cxx.o
/home/r/itk/ITK/Modules/Remote/MGHIO/src/itkMGHImageIOFactory.cxx:19:10: fatal error: itkMGHImageIOFactory.h: No such file or directory
   19 | #include "itkMGHImageIOFactory.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/CMake/itkCheckForOrphanedRemoteModules.cmake
+++ b/CMake/itkCheckForOrphanedRemoteModules.cmake
@@ -1,0 +1,52 @@
+#
+# Detect remote-module clones left in the source tree without a controlling
+# <name>.remote.cmake. Such orphans are discovered by the module DAG glob and
+# break the build, commonly after a module is ingested into ITK core.
+function(itk_check_for_orphaned_remote_modules)
+  set(_remote_dir "${ITK_SOURCE_DIR}/Modules/Remote")
+  file(GLOB _entries CONFIGURE_DEPENDS "${_remote_dir}/*")
+
+  set(_orphans)
+  foreach(_entry ${_entries})
+    if(NOT IS_DIRECTORY "${_entry}")
+      continue()
+    endif()
+    get_filename_component(_name "${_entry}" NAME)
+    if(_name STREQUAL "Deprecated")
+      continue()
+    endif()
+    if(
+      NOT
+        EXISTS
+          "${_remote_dir}/${_name}.remote.cmake"
+      AND
+        NOT
+          EXISTS
+            "${_remote_dir}/Deprecated/${_name}.remote.cmake"
+    )
+      list(APPEND _orphans "${_entry}")
+    endif()
+  endforeach()
+
+  if(_orphans)
+    set(_rm_lines)
+    foreach(_orphan ${_orphans})
+      if(CMAKE_HOST_WIN32)
+        string(APPEND _rm_lines "    rmdir /s /q \"${_orphan}\"\n")
+      else()
+        string(APPEND _rm_lines "    rm -rf \"${_orphan}\"\n")
+      endif()
+    endforeach()
+    message(
+      FATAL_ERROR
+      "Stale (orphaned) remote-module source directories detected.\n"
+      "These were fetched into the source tree but no longer have a "
+      "controlling <name>.remote.cmake (the module was likely ingested into "
+      "ITK core). They are picked up by the module dependency scan and will "
+      "break the build. Delete them and re-configure:\n\n"
+      "${_rm_lines}"
+    )
+  endif()
+endfunction()
+
+itk_check_for_orphaned_remote_modules()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -769,6 +769,11 @@ if(BUILD_TESTING OR ITK_BUILD_DOCUMENTATION OR ITK_WRAP_PYTHON)
 endif()
 
 #----------------------------------------------------------------------
+# Fail early on stale remote-module clones left without a controlling
+# .remote.cmake (e.g. after ingestion into core).
+include(itkCheckForOrphanedRemoteModules)
+
+#----------------------------------------------------------------------
 # Make sure remote modules are downloaded before sorting out the module
 # dependencies.
 add_subdirectory(Modules/Remote)

